src/test/java/book/shop/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,222 @@
+    @Sql(scripts = "classpath:database/book/delete-all-book&category.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Creat category")
+    void creatCategory_Ok() throws Exception {
+        CreateCategoryRequestDto categoryRequestDto = new CreateCategoryRequestDto();
+        categoryRequestDto.setName("Category name");
+        categoryRequestDto.setDescription("Category description");
+        String request = objectMapper.writeValueAsString(categoryRequestDto);
+        MvcResult mvcResult = mockMvc.perform(post("/categories")
+                .content(request)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+        CategoryDto expected = new CategoryDto()
+                .setId(3L)
+                .setName(categoryRequestDto.getName())
+                .setDescription(categoryRequestDto.getDescription());
+        CategoryDto actual = objectMapper
+                .readValue(mvcResult.getResponse()
+                        .getContentAsString(), CategoryDto.class);
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(scripts = "classpath:database/book/add-books-and-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/book/delete-all-book&category.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Update category")
+    void updateCategory_Ok() throws Exception {
+        Long categoryId = 1L;
+        CreateCategoryRequestDto requestDto = new CreateCategoryRequestDto()
+                .setName("Category name")
+                .setDescription("Category description");
+        String request = objectMapper.writeValueAsString(requestDto);
+        MvcResult mvcResult = mockMvc.perform(put("/categories/{id}",categoryId)
+                .content(request)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        CategoryDto expected = new CategoryDto()
+                .setId(categoryId)
+                .setName(requestDto.getName())
+                .setDescription(requestDto.getDescription());
+        CategoryDto actual = objectMapper.readValue(mvcResult.getResponse()
+                .getContentAsString(), CategoryDto.class);
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(scripts = "classpath:database/book/add-books-and-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/book/delete-all-book&category.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
```

Comment:
Test methods are overwhelmed with annotations.
Please try to arrange tests in way so you execute one filling script in BeforeAll method and then make cleanup in AfterAll method